### PR TITLE
Support events with slugs

### LINKED ISSUE #384: Support event slug as deeplink
### What
Slug urls to an event should also open in the app.

### Why
App!

### PR DESCRIPTION
**Pull request by JeeVee11**
Closes #384.

### Summary
Adds support for events with slugs as deeplink.
As a consequence of this EventCubit and RegistrationCubit have also been merged.

### How to test
Steps to test the changes you made:
1. Go to staging.thalia.nu/events/(some slug).
2. See that the event loads correctely.
3. If there are no registrations, it does not show an error.

### CODE REVIEW

File: lib/ui/screens/event_screen.dart [hunk 1]
```diff
@@ -16,10 +16,11 @@ import 'package:url_launcher/url_launcher.dart';
 import 'package:reaxit/config.dart' as config;
 
 class EventScreen extends StatefulWidget {
-  final int pk;
+  final String? slug;
   final Event? event;
+  final int? pk;
 
-  EventScreen({required this.pk, this.event}) : super(key: ValueKey(pk));
+  const EventScreen({required this.pk, required this.slug, this.event});
```
Comment (JAicewizard) [inline lib/ui/screens/event_screen.dart:23]: This needs an assert that not both slug and pk can be null. The `required` is also no longer neccesary

File: test/mocks.mocks.dart [hunk 2]
```diff
@@ -334,7 +334,7 @@ class MockApiRepository extends _i1.Mock implements _i4.ApiRepository {
         returnValueForMissingStub: null,
       );
   @override
-  _i6.Future<_i3.Event> getEvent({required int? pk}) => (super.noSuchMethod(
+  _i6.Future<_i3.Event> getEventPk({required int? pk}) => (super.noSuchMethod(
```
Comment (JAicewizard) [inline test/mocks.mocks.dart:337]: This needs to be regenerated somewhere. @DeD1rk

**Review by JAicewizard — CHANGES_REQUESTED** [review]

File: test/mocks.mocks.dart (hunk 2 above)
Comment (DeD1rk) [reply test/mocks.mocks.dart:337]: I think thats just the `flutter pub run build_runner build` thing

**Comment by JAicewizard** [comment]
looks good, I will try it out in about an hour

**Comment by JAicewizard** [comment]
If possible the share button should share the slug as well. RN it still shares the pk

**Comment by JAicewizard** [comment]
nice works

**Pull request merged**